File: src/tests/Behavioral/PackageVarFuncLitTypeLift/varfirst.go
// The SAME root as main.go's `makers`, in the arrangement that produced the other symptom.
//
// Which symptom appeared depended only on declaration order within the FILE: with no function
// declaration visited before the var, currentFuncPrefix was not merely stale but nil, and writing
// the lifted declaration into it panicked (nil receiver inside strings.Builder.copyCheck). The
// converter recovers that panic per file, so the whole FILE was skipped with only a
// "visit file error" warning -- a silent, total loss of this file's conversion.
//
// This file therefore declares the var FIRST, before any function declaration.
package main

var varFirst = func(s string) Greeter {
	return struct{ Greeter }{namedGreeter{s}}
}("package-scope")

var varFirstLabel = "varfirst"

File: src/tests/Behavioral/PackageVarFuncLitTypeLift/varfirst.cs
namespace go;

partial class main_package {


[GoType("dyn")] internal partial struct varFirst_type {
    public Greeter Greeter;
}
internal static Greeter varFirst = ((Func<@string, Greeter>)(s => {
    return new varFirst_type(new namedGreeter(s));
}))("package-scope");

internal static @string varFirstLabel = "varfirst"u8;

} // end main_package
